### PR TITLE
Show focus on Submenu item for keyboard access

### PR DESCRIPTION
**Pull request by chiuam**
### Platforms Impacted
- [ ] iOS
- [X] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

This PR refines the focus behavior for Submenu item to align with the native experience. 

1. When opening a menu(via keyboard/mouse), the first menu item should not get focus. 
            - Covered.
2. When opening a submenu via mouse, the first menu item should not get focus. 
            - Regressed by this change, created a task for followup.
3. When opening a submenu via keyboard, the first menu item should get focus and the parent submenu item should lose focus/ fades to a gray color background.
            - This change achieves the first part, a task has been filed to followup on the second part.

### Verification

Tested Submenus on win32, didn't notice any behavioral changes. 

Before:

https://user-images.githubusercontent.com/67026167/211050743-78ec34af-8fea-4c57-aa58-9d31324f49bd.mov

After:

https://user-images.githubusercontent.com/67026167/211050712-02f2f912-8c40-4f60-823e-2693f08e284a.mov

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [X] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Menu/src/MenuList/MenuList.tsx [hunk 1]
```diff
@@ -39,6 +40,7 @@ export const MenuList = compose<MenuListType>({
   },
   useRender: (userProps: MenuListProps, useSlots: UseSlots<MenuListType>) => {
     const menuList = useMenuList(userProps);
+    const context = useMenuContext();
     const contextValue = useMenuListContextValue(menuList);
```
Comment (Saadnajmi) [inline packages/components/Menu/src/MenuList/MenuList.tsx:44]: nit, rename these maybe so they are more clear?
```suggestion
    const menuContext = useMenuContext();
    const menuListContextValue = useMenuListContextValue(menuList);
```

**Review by Saadnajmi — APPROVED** [review]
Looks good to me, I think I'd want to check with win32 to make sure there's no regression.

File: packages/components/Menu/src/MenuList/MenuList.tsx [hunk 2]
```diff
@@ -56,8 +58,7 @@ export const MenuList = compose<MenuListType>({
               <Slots.focusZone
                 componentRef={focusZoneRef}
                 focusZoneDirection={'vertical'}
-                defaultTabbableElement={focusZoneRef}
-                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                defaultTabbableElement={context.isSubmenu ? null : focusZoneRef} // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (Saadnajmi) [inline packages/components/Menu/src/MenuList/MenuList.tsx:61]: Could we add a comment saying why setting `defaultTabbableElement` affects which menu item gets focused?

**Comment by chiuam** [comment]
> Looks good to me, I think I'd want to check with win32 to make sure there's no regression.

Yep, tested a few submenu scenarios before & after my change, didn't notice any behavioral changes. I did find a redbox error complaining about Callout shadow node, though I made sure my change wasn't the cause.

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**